packages/react/__tests__/src/components/LoaderOverlay/index.js

```diff
@@ -37,6 +37,44 @@ test('handles variants', () => {
   expect(largeNode.classList.contains('Loader__overlay--large')).toBe(true);
 });
 
+test('handles focus', () => {
+  const loaderOverlay = mount(
+    <LoaderOverlay className="baz" role="alert" label="loading" focus>
+      Some text
+    </LoaderOverlay>
+  );
+
+  setTimeout(() => {
+    expect(document.activeElement).toBe(loaderOverlay.getDOMNode());
+  });
+});
+
+test('handles not being focused', () => {
+  const loaderOverlay = mount(
+    <LoaderOverlay className="baz" role="alert" label="loading">
+      Some text
+    </LoaderOverlay>
+  );
+
+  setTimeout(() => {
+    expect(document.activeElement).not.toBe(loaderOverlay.getDOMNode());
+  });
+});
+
+test('handles being passed a ref', () => {
+  const loaderRef = React.createRef(null);
+
+  const loaderOverlay = mount(
+    <LoaderOverlay className="baz" role="alert" label="loading" ref={loaderRef}>
+      Some text
+    </LoaderOverlay>
+  );
+
+  setTimeout(() => {
+    expect(document.activeElement).toStrictEqual(loaderOverlay.getDOMNode());
+  });
+});
+
```

Comment:
Test cases above this test for implementation of the new `focus` prop, but also added in a test to ensure backwards compatibility. As it stands, if we were to remove backwards compatibility for use of `forwardRef`, it would be a breaking change for current implementations where that prop is used.